File: shanny-blog-java/blog-pojo/src/main/java/com/shanny/entity/Media.java
package com.shanny.entity;

import com.shanny.converter.LongListConverter;
import com.shanny.enums.MediaEnum;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "medias")
public class Media {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(length = 50, nullable = false)
    private String title;

    @Column(length = 300)
    private String content;

    @Column(length = 300)
    private String image;

    @NotNull
    @Column(length = 300, nullable = false)
    private String href;

    @Convert(converter = LongListConverter.class)
    private List<Long> tags;

    @NotNull
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private MediaEnum.MediaType type;

    @NotNull
    @Column(nullable = false)
    private Boolean published = true;

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Column(nullable = false)
    private LocalDateTime createTime;

    @NotNull
    @DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @Column(nullable = false)
    private LocalDateTime updateTime;
}

File: shanny-blog-net/blog-pojo/Entities/Media.cs
using blog_common.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace blog_pojo.Entities
{
    [Table("medias")]
    public class Media
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [MaxLength(50)]
        [Required]
        [Column("title")]
        public string Title { get; set; } = string.Empty;

        [MaxLength(300)]
        [Column("content")]
        public string Content { get; set; } = string.Empty;

        [MaxLength(300)]
        [Column("image")]
        public string Image { get; set; } = string.Empty;

        [MaxLength(300)]
        [Required]
        [Column("href")]
        public string Href { get; set; } = string.Empty;

        [Column("tags")]
        public List<long> Tags { get; set; } = new();

        [Required]
        [Column("type")]
        public MediaType Type { get; set; }

        [Required]
        [Column("published")]
        public bool Published { get; set; } = true;

        [Required]
        [Column("createTime")]
        public DateTime CreateTime { get; set; }

        [Required]
        [Column("updateTime")]
        public DateTime UpdateTime { get; set; }
    }
}
